packages/console/src/components/CreateTenantModal/SelectTenantPlanModal/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { type TenantInfo } from '@logto/schemas/models';
+import { toast } from 'react-hot-toast';
+import { Trans, useTranslation } from 'react-i18next';
+import Modal from 'react-modal';
+
+import { useCloudApi } from '@/cloud/hooks/use-cloud-api';
+import { ReservedPlanId } from '@/consts/subscriptions';
+import DangerousRaw from '@/ds-components/DangerousRaw';
+import ModalLayout from '@/ds-components/ModalLayout';
+import TextLink from '@/ds-components/TextLink';
+import useSubscriptionPlans from '@/hooks/use-subscription-plans';
+import * as modalStyles from '@/scss/modal.module.scss';
+import { type SubscriptionPlan } from '@/types/subscriptions';
+
+import { type CreateTenantData } from '../type';
+
+import PlanCardItem from './PlanCardItem';
+import * as styles from './index.module.scss';
+
+type Props = {
+  createTenantData?: CreateTenantData;
+  onClose: (tenant?: TenantInfo) => void;
+};
+
+function SelectTenantPlanModal({ createTenantData, onClose }: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const { data: subscriptionPlans } = useSubscriptionPlans();
+  const cloudApi = useCloudApi();
+  if (!subscriptionPlans || !createTenantData) {
+    return null;
+  }
+
+  const handleSelectPlan = async (plan: SubscriptionPlan) => {
+    const { id: planId } = plan;
+    if (planId === ReservedPlanId.free) {
+      try {
+        const { name, tag } = createTenantData;
+        const newTenant = await cloudApi.post('/api/tenants', { body: { name, tag } });
+
+        onClose(newTenant);
+      } catch (error: unknown) {
+        toast.error(error instanceof Error ? error.message : String(error));
+      }
+    }
+    // Todo @xiaoyijun implement checkout
+  };
+
+  return (
+    <Modal
+      shouldCloseOnEsc
+      isOpen={Boolean(createTenantData)}
+      className={modalStyles.content}
+      overlayClassName={modalStyles.overlay}
+      onRequestClose={() => {
+        onClose();
+      }}
+    >
+      <ModalLayout
+        title="upsell.create_tenant.title"
+        subtitle={
+          <DangerousRaw>
+            <Trans
+              components={{
+                a: <TextLink href="https://logto.io/pricing" target="_blank" />,
+              }}
+            >
+              {t('upsell.create_tenant.description')}
+            </Trans>
+          </DangerousRaw>
+        }
+        size="xlarge"
+        onClose={onClose}
+      >
+        <div className={styles.container}>
+          {subscriptionPlans.map((plan) => (
+            <PlanCardItem
+              key={plan.id}
+              plan={plan}
+              onSelect={async () => {
+                await handleSelectPlan(plan);
+              }}
```

Comment:
no need to `await`?

Reply:
We have async logic in the `handleSelectPlan`.